13-dplyr.Rmd

```diff
@@ -0,0 +1,191 @@
+
+Here we're going to cover 6  of the most commonly used functions as well as using pipes (`%>%`) to combine them. 
+
+1. `select()`
+2. `filter()`
+3. `group_by()`
+4. `summarize()`
+5. `mutate()`
+
+If you have have not installed this package earlier, please do so:
+
+```{r,eval=FALSE}
+install.packages('dplyr')
+```
+
+Now let's load the package:
+
+```{r,message=FALSE}
+library(dplyr)
+```
+
+#### Using select()
+
+If, for example, we wanted to move forward with only a few of the variables in our dataframe we could use the `select` function. This will keep only the variables you select.
+
+```{r}
+year_country_gdp <- select(gapminder,year,country,gdpPercap)
+```
+
+> draw this on the board? I'd like to include a diagram here too
+
+If we open up `year_country_gdp` we'll see that it only contains the year, country and gdpPercap. Above we used 'normal' grammar, but `dplyr` strengths lie in combining several functions using pipes. Since the pipes grammar is unlike anything we've seen in R before, let's repeat what we've done above using pipes.
+
+```{r}
+year_country_gdp <- gapminder %>% select(year,country,gdpPercap)
+```
+
+> draw this on the board? I'd like to include a diagram here too
+
+To help you understand why we wrote that in that way, let's walk through it step by step. First we summon the gapminder dataframe and pass it on, using the pipe symbol `%>%`, to the next step which is the select function. In this case we don't specify which data object we use in the `select` function since in gets that from the previous pipe.
+
+#### Using filter()
+
+If we now wanted to move forward with the above, but only with European countries, we can combine `select` and `filter`
+
+```{r}
+year_country_gdp_euro <- gapminder %>%
+    filter(continent=="Europe") %>%
+    select(year,country,gdpPercap)
+```
+
+> #### Challenge 1 {.challenge}
+>
+> Can you write a single command (which can span multiple lines and includes pipes) that will produce a dataframe that has the African values for `lifeExp`, `country` and `year`, but not for other Continents. 
+>How many rows does your dataframe have and why?
+>
+
+
+As with last time, first we pass the gapminder dataframe to the filter function, then we pass the filtered version of the gapminder dataframe to the select function. **Note:** The order of operations is very important in this case. If we used 'select' first, filter would not be able to find the variable continent since we would have removed it in the previous step.
+
+#### Using group_by()
+
+Now, I told you all that I would be reducing the error prone repetitiveness of what we can do with base R, but up to now we haven't done that since we would have to repeat the above for each continent. Instead of `filter`, which will only pass observations that meet your criteria (in the above: `continent=="Europe"`), we can use `group_by`, which will essentially use every unique criteria that you could have used in filter.
```

Comment:
Should be written not as a transcript to be read, but as a description of the information to be taught.  I think also perhaps these few sentences could be reworded for clarity. I think this is also true for the explanation on line 102. The `group_by()` function is a bit of a tricky concept to explain well and so could potentially use a bit more space to explain?